docs/releases/uv-upgrade.md

```diff
@@ -0,0 +1,317 @@
+    my_site_views = my_site.views:create_blueprint
+invenio_assets.webpack =
+    my_site_theme = my_site.webpack:theme
+# ...other entry points for Celery tasks, models, CLI commands, etc.
+```
+
+#### After: `site/pyproject.toml`
+
+```toml title="site/pyproject.toml"
+[project]
+name = "my-site" # (1)!
+version = "1.0.0" # (2)!
+description = "My Site customizations for Invenio RDM."
+# (3)!
+
+[project.entry-points."invenio_base.blueprints"] # (4)!
+my_site_views = "my_site.views:create_blueprint"
+[project.entry-points."invenio_assets.webpack"]
+my_site_theme = "my_site.webpack:theme"
+# ...other entry points for Celery tasks, models, CLI commands, etc.
+
+[build-system] # (5)!
+requires = ["hatchling"]
+build-backend = "hatchling.build"
+```
+
+1. Package name should match what's referenced in the root `pyproject.toml`
+2. Version is a required field in `pyproject.toml`
+3. We don't need to define any dependencies here, since they are managed at the root `pyproject.toml`
+4. Entry points for blueprints, assets, Celery tasks, etc. remain similar
+5. Modern build system - `hatchling` is recommended for pure Python packages
+
+### Update Invenio configuration 📜
+
+Update the `.invenio` configuration file so that `invenio-cli` uses `uv` commands instead of `pipenv` for dependency management:
+
+```ini title=".invenio" hl_lines="4"
+[cli]
+flavour = RDM
+logfile = /logs/invenio-cli.log
+python_package_manager = uv
+
+[cookiecutter]
+project_name = My Site
+...
+```
+
+### Clean up old files 📜
+
+Remove any old files:
+
+```bash
+rm Pipfile Pipfile.lock site/setup.cfg site/setup.py site/MANIFEST.in
+```
+
+### Generate new Python dependencies lockfile
+
+Generate the `uv.lock` file:
+
+```bash
+invenio-cli packages lock
+```
+
+### Update Dockerfile
+
+#### Before: Docker with Pipenv
+
+```dockerfile title="Dockerfile"
+...
+COPY site ./site
+COPY Pipfile Pipfile.lock ./
+RUN pipenv install --deploy --system
+...
+```
+
+#### After: Docker with uv
+
+```dockerfile title="Dockerfile"
+...
+# Python and uv configuration
+ENV PYTHONDONTWRITEBYTECODE=1 \
+    PYTHONUNBUFFERED=1 \
+    # Cache directory for uv's package downloads - persisted across builds with Docker BuildKit
+    UV_CACHE_DIR=/opt/.cache/uv \
+    # Pre-compile Python bytecode for faster startup times in production
+    UV_COMPILE_BYTECODE=1 \
+    # Strictly use versions from uv.lock file, failing if lock file is outdated
+    UV_FROZEN=1 \
+    # Copy packages instead of symlinking - required for Docker's layered filesystem
+    UV_LINK_MODE=copy \
+    # Use the system's Python installation instead of uv managing Python versions
+    UV_NO_MANAGED_PYTHON=1 \
+    UV_SYSTEM_PYTHON=1 \
+    UV_PROJECT_ENVIRONMENT=/usr/ \
+    UV_PYTHON_DOWNLOADS=never \
+    # Require and verify package hashes match those in the lock file
+    UV_REQUIRE_HASHES=1 \
+    UV_VERIFY_HASHES=1
+
+# Copy uv binary from official image
+COPY --from=ghcr.io/astral-sh/uv:latest /uv /usr/local/bin/uv
```

Comment:
just as a note, AFAIK the `uv` image is based on alpine (which uses `musl` instead of `glibc`) – which should **not** be an issue here because `uv` is rust/statically compiled (and thus doesn't need to be compiled against `glibc` vs. `musl`)

it might cause trouble if a dynamically linked binary were copied over to a `glibc`-based image